packages/editor-ui/src/components/ParameterInputFull.vue

```diff
@@ -223,12 +223,30 @@ export default mixins(showMessage).extend({
 			}
 		},
 		onDrop(data: string) {
-			this.forceShowExpression = true;
+			const useDataPath = !!this.parameter.requiresDataPath && data.startsWith('{{ $json');
+			if (!useDataPath) {
+				this.forceShowExpression = true;
+			}
 			setTimeout(() => {
 				if (this.node) {
 					const prevValue = this.isResourceLocator ? this.value.value : this.value;
 					let updatedValue: string;
-					if (typeof prevValue === 'string' && prevValue.startsWith('=') && prevValue.length > 1) {
+					if (useDataPath) {
+						let newValue = data.replace('{{ $json', '').replace(new RegExp('}}$'), '').trim();
```

Comment:
kind of a hack - ideally would construct this from a data object instead.. but would require too much refactoring of mapping